app/src/test/java/org/mozilla/focus/session/SessionManagerTest.java

```diff
@@ -166,7 +166,7 @@ public void testShareIntentViaNewIntent() {
 
         assertTrue(sessionManager.hasSession());
     }
-
+    @Test (expected = IllegalStateException.class)
```

Comment:
We do not expect this exception here. Can you post the stacktrace you are seeing in the issue? We can then figure out what goes wrong in this test case..

Reply:
My apologies for the late reply @pocmo. Here is the stacktrace when adding just the @Test annotation:

java.lang.IllegalStateException: Attempting to retrieve search engines without a corresponding init()

	at org.mozilla.focus.search.SearchEngineManager.awaitLoadingSearchEnginesLocked(SearchEngineManager.java:317)
	at org.mozilla.focus.search.SearchEngineManager.getDefaultSearchEngine(SearchEngineManager.java:298)
	at org.mozilla.focus.utils.UrlUtils.createSearchUrl(UrlUtils.java:79)
	at org.mozilla.focus.session.SessionManager.createSessionFromIntent(SessionManager.java:94)
	at org.mozilla.focus.session.SessionManager.handleNewIntent(SessionManager.java:67)
	at org.mozilla.focus.session.SessionManagerTest.testShareIntentWithTextViaNewIntent(SessionManagerTest.java:177)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:523)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:226)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:108)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:35)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.robolectric.internal.SandboxTestRunner$1.evaluate(SandboxTestRunner.java:62)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)

Reply:
Ah, SearchEngineManager seems to have problems loading the search engine configuration files. We recently updated Robolectric and I think this should work now. You could try this again on top of the latest 'master' state.

Reply:
Huzzah! It now no longer return an illegal state exception, however, there is now a junit comparrison failure on line:186 assertEquals(TEST_URL, session.getUrl().getValue());

Expected :https://github.com/mozilla-mobile/focus-android
Actual   :https://www.google.com/search?q=Hello%20World%20Focus&ie=utf-8&oe=utf-8&client=firefox-b-1

Reply:
@berlinbra Ah, yeah, I think our expectation is wrong here. After all this Intent should indeed trigger a search. If you update the expected URL here then this should be a patch that we can land. :) 👍